test/test_backend.py

```diff
@@ -170,7 +170,18 @@ def test_counts_over_time(self):
                     # Part 2: all other row counts should be non-zero
                     self.assertGreater(df[col][row], 0, msg=f"Table '{row}' had 0 rows in run '{col}'")
 
-
+    def test_subgraph(self):
+        "tests subgraphs"
+        ids = [1738170, 1738171, 1738202, 1738203]
+        edges = subgraph(ids)
+        #Part 1: Test that the edge list is the right size
+        self.assertEqual(len(ids)-1,len(edges))
+        #Part 2: Test the edges have the right valus
+        for i in range(len(ids)-1):
+            self.assertEqual(edges[i][0],ids[i])
+        for i in range(1,len(ids)):
+            self.assertEqual(edges[i-1][1],ids[i])
```

Comment:
Hey @hugwuoke, this is fancier than I imagined, which was something like:

    self.assertEqual(edges, [
        [
          "1738170",
          "1738171"
        ],
        [
          "1738170",
          "1738202"
        ],
        [
          "1738170",
          "1738203"
        ]
      ])

Your approach works for this specific example where there happens to be a single root and the other three nodes are direct children of that root. But, you'll see, I just added more things that the test should be able to do, and for those, this approach won't work.